lib/charms/tempo_k8s/v2/tracing.py

```diff
@@ -289,27 +308,79 @@ def dump(self, databag: Optional[MutableMapping] = None, clear: bool = True):
+if int(pydantic.version.VERSION.split(".")[0]) < 2:
 
-    protocol: ReceiverProtocol
-    port: int
+    class ProtocolType(BaseModel):  # type: ignore
+        """Protocol Type."""
 
+        class Config:
+            """Pydantic config."""
 
-class TracingProviderAppData(DatabagModel):  # noqa: D101
-    """Application databag model for the tracing provider."""
+            use_enum_values = True
+            """Allow serializing enum values."""
 
-    host: str
-    """Server hostname (local fqdn)."""
+        name: str = Field(
+            ...,
+            description="Receiver protocol name. What protocols are supported (and what they are called) "
+            "may differ per provider.",
+            examples=["otlp_grpc", "otlp_http", "tempo_http"],
+        )
+
+        type: TransportProtocolType = Field(
+            ...,
+            description="The transport protocol used by this receiver.",
+            examples=["http", "grpc"],
+        )
+
+else:
 
-    receivers: List[Receiver]
-    """Enabled receivers and ports at which they are listening."""
+    class ProtocolType(BaseModel):
+        """Protocol Type."""
 
-    external_url: Optional[str] = None
-    """Server url. If an ingress is present, it will be the ingress address."""
+        model_config = ConfigDict(
+            # Allow serializing enum values.
+            use_enum_values=True
+        )
+        """Pydantic config."""
 
-    internal_scheme: Optional[str] = None
-    """Scheme for internal communication. If it is present, it will be protocol accepted by the provider."""
+        name: str = Field(
+            ...,
+            description="Receiver protocol name. What protocols are supported (and what they are called) "
+            "may differ per provider.",
+            examples=["otlp_grpc", "otlp_http", "tempo_http"],
+        )
+
+        type: TransportProtocolType = Field(
+            ...,
+            description="The transport protocol used by this receiver.",
+            examples=["http", "grpc"],
+        )
+
+
+class Receiver(BaseModel):
+    """Specification of an active receiver."""
+
+    protocol: ProtocolType = Field(..., description="Receiver protocol name and type.")
+    url: str = Field(
+        ...,
+        description="URL at which the receiver is reachable. If there's an ingress, it would be the external URL. Otherwise, it would be the service's fqdn or internal IP",
+        examples=[
+            "http://traefik_address:2331",
+            "https://traefik_address:2331",
+            "http://tempo_public_ip:2331",
+            "https://tempo_public_ip:2331",
+            "tempo_public_ip:2331",
+        ],
+    )
+
+
+class TracingProviderAppData(DatabagModel):  # noqa: D101
+    """Application databag model for the tracing provider."""
+
+    receivers: List[Receiver] = Field(
+        ...,
+        description="A list of enabled receivers in the form of the protocol they use and their resolvable server url.",
```

Comment:
The form is already defined in the models, no need to reword it here

```suggestion
        description="List of all receivers enabled on the tracing provider.",
```